core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/HeartBeatModule.java

```diff
@@ -0,0 +1,185 @@
+package com.microsoft.applicationinsights.internal.heartbeat;
+
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.extensibility.TelemetryModule;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+/**
+ * <h1>HeartBeat Provider Module</h1>
+ *
+ * <p>
+ *   This module is the core module which is is configured by default with default settings
+ *   when ApplicationInsights SDK boots up. It is used to transmit diagnostic heartbeats to
+ *   Application Insights backend.
+ * </p>
+ *
+ * @author Dhaval Doshi
+ * @since 03-30-2018
+ */
+public class HeartBeatModule implements TelemetryModule {
```

Comment:
It appears this needs a no arg constructor as well.

Reply:
We can have that ... but not needed. We can pass the null inside the parameterized constructor. There is a check there which handles it. I don't think we need to expose other constructor here if we can avoid.